### PR TITLE
Hattie/forms integration

### PR DESCRIPTION
**Pull request by hllavina**
## Status
_Use labels (`review needed`, `in progress`, or `paused`) to declare status_

#### What does this PR do?
Adds page-forms template, adds styles for forms, integrates login functions script

#### How can a reviewer manually see the effects of these changes?
Test server, document uploaded to Jira ticket with specific links

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/NTI-147

#### Todo:
- [ ] Documentation
- [ ] Stakeholder approval

#### Requires new or updated plugins, themes, or libraries?
YES  - contact form 7 plugin

#### Requires change to deploy process?
NO

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -1462,3 +1462,73 @@ a.comment-edit-link:hover {
...
+    border: 1px solid #338bc5;
+    border-radius: 3px;
+    padding: 5px 10px;
+    background-color: #338bc5;
+    background-image: none;
+    font-size: 1rem;
+    color: #fff;
+    margin-top:10px;
+}
+.wpcf7 label {
```
Comment (frrrances) [inline css/style.css]: I think the labels should stay bold so this rule can be removed.

File: css/style.css [hunk 2]
```diff
@@ -1462,3 +1462,73 @@ a.comment-edit-link:hover {
...
+}
+
+.wpcf7  legend {
+	padding: 0 1rem;
+	font-size: .9rem;
+	color: #595959;
+	text-transform: uppercase;
+}
+
+.button-primary.green:focus, .button-primary.green:hover, .wrap-outer-header-local .green.action-auth:focus, .wrap-outer-header-local .green.action-auth:hover {
```
Comment (frrrances) [inline css/style.css]: I didn't see a form with these styles (and the ones below) - Can you let me know what they're for?

Comment (hllavina) [reply css/style.css]: Thanks, that was for the 'auto fill' button, I will fix it. I think we can use the plugin  to develop the mit-only forms  and what pages they will be placed on will have the auth, I will have a recommendation at the meeting . thanks for the review!

Comment (frrrances) [reply css/style.css]: Okay, sounds good!

**Comment by matt-bernhardt** [comment]
Hey, I wanted to let you know that I'm working on a review of these code changes. Hoping to have it done today.

**Comment by hllavina** [comment]
yes that is a good point. i was wondering if the forms should  have a slim header at all, I think the pages they are on should look the other pages and should have the navigation. this is a question I wanted to raise

**Comment by matt-bernhardt** [comment]
Agreed - consistency across content is A Good Thing. We can talk next week about whether it makes sense to leave the slim heading place here, or convert over to the standard header?

Other questions:

1) A lot of the CodeClimate and Travis failures boil down to two causes: the addition of `js/login_functions.js` (which is already on the site), and `lib/hours.php` which is the subject of a separate PR at MITLibraries/MITlibraries-parent#227. A few of them however, do seem worth addressing?

Have you attempted to clean `js/login_functions.js`? Is it just a horrible time sink (as I expect it would be)?

2) It might be worth talking about how we want the theme to conditionally load page-specific scripts and styles. This PR uses a page template, while elsewhere in the theme we're checking for specific pages in `functions.php` and loading content that way. I waffle on the best approach, but whatever we pick I think using it consistently would be good.

**Comment by hllavina** [comment]
Thanks for your thoughts, I will clean login_functions. when you say that it is already in the site, do you mean as part of the current forms  deployment under scripts? but it is not part of the wordpress theme. I will create ticket to remove all the forms files when this solution is complete and deployed to production 
In the past I've loaded javascript only on the templates that needed it. I did it in functions.php because that seemed the preferred way in the theme, I agree it would be good to discuss

**Comment by matt-bernhardt** [comment]
This all sounds good - looking forward to our meeting later this week.

**Comment by hllavina** [comment]
I've deployed the latest branch to test with these comments incorporated, please feel free to post more feedback here or in the jira ticket https://mitlibraries.atlassian.net/browse/NTI-147

File: css/style.css [hunk 3]
```diff
@@ -1462,3 +1462,62 @@ a.comment-edit-link:hover {
...
+	width:75%;
+}
+.wpcf7  legend {
+	padding: 0 1rem;
+	font-size: .9rem;
+	color: #595959;
+	text-transform: uppercase;
+}
+
+.wpcf7  .button-primary.green:focus, .button-primary.green:hover, .wrap-outer-header-local .green.action-auth:focus, .wrap-outer-header-local .green.action-auth:hover {
```
Comment (frrrances) [inline css/style.css]: Can you add the scoping class `.wpcf7` to all the selectors in this rule? Especially, it looks like the second selector would apply to any `button-primary green` (I'm not sure whether the other selectors are unique to forms.)

File: css/style.css [hunk 4]
```diff
@@ -1462,3 +1462,62 @@ a.comment-edit-link:hover {
...
+
+    background-color: #2ea76a;
+    border-color: #43926a;
+
+}
+.wpcf7  .button-primary.green, .wrap-outer-header-local .green.action-auth {
```
Comment (frrrances) [inline css/style.css]: Can you add the scoping class `.wpcf7` to the second selector in this rule? (I'm not sure whether it is unique to forms.)

**Comment by hllavina** [comment]
ok, thank you for the review

**Pull request merged**